src/managers/OnOfframpManager.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {CastLib} from "src/misc/libraries/CastLib.sol";
+import {IERC165} from "src/misc/interfaces/IERC165.sol";
+import {SafeTransferLib} from "src/misc/libraries/SafeTransferLib.sol";
+
+import {PoolId} from "src/common/types/PoolId.sol";
+import {ShareClassId} from "src/common/types/ShareClassId.sol";
+import {UpdateContractType, UpdateContractMessageLib} from "src/spoke/libraries/UpdateContractMessageLib.sol";
+
+import {IBalanceSheet} from "src/spoke/interfaces/IBalanceSheet.sol";
+import {IUpdateContract} from "src/spoke/interfaces/IUpdateContract.sol";
+
+import {IOnOfframpManager} from "src/managers/interfaces/IOnOfframpManager.sol";
+import {IOnOfframpManagerFactory} from "src/managers/interfaces/IOnOfframpManagerFactory.sol";
+import {IDepositManager, IWithdrawManager} from "src/managers/interfaces/IBalanceSheetManager.sol";
+
+/// @title  OnOfframpManager
+/// @notice Balance sheet manager for depositing and withdrawing ERC20 assets.
+///         - Onramping is permissionless: once an asset is allowed to be onramped and ERC20 assets have been
+///           transferred to the manager, anyone can trigger the balance sheet deposit.
+///         - Offramping is permissioned: only predefined relayers can trigger withdrawals to predefined
+///           offramp accounts.
+contract OnOfframpManager is IOnOfframpManager {
+    using CastLib for *;
+
+    PoolId public immutable poolId;
+    address public immutable spoke;
+    ShareClassId public immutable scId;
+    IBalanceSheet public immutable balanceSheet;
+
+    mapping(address asset => bool) public onramp;
+    mapping(address asset => mapping(address relayer => address receiver)) public offramp;
```

Comment:
QUESTION 

would map as `mapping(address asset => address receiver)` offramp and as `mapping(address relayer => bool)` relayer

Reply:
Why? Would just be 2 storage reads on every transaction.

Reply:
But whitelisting a relayer for every asset is really, really annoying.

Reply:
That's a really good point! Will change.